Watchman.Integrations/MongoDB/Session.cs

```diff
@@ -54,7 +55,12 @@ public Session(IMongoDatabase database)
 
         public async Task DeleteAsync<T>(T entity) where T : Entity
         {
-            await this.GetCollection<T>().DeleteOneAsync(x => x.Id == entity.Id);
+            await this.GetCollection<T>().UpdateOneAsync(x => x.Id == entity.Id, Builders<T>.Update.Set(p => p.IsDeleted, true));
```

Comment:
zamiast ustawiania IsDeleted na true

możesz użyć po prostu entity.Delete();

Reply:
Jedyny sposób jaki na razie udało mi się znaleźć żeby tak zrobić, to wyciągnąć kolekcję i zrobić na niej select po przefiltrowaniu i zapisać od nowa. Inaczej albo się nie da albo nie znalazłem w docach mongoDb z c#

Reply:
ale po co ty chcesz wyciągać cokolwiek z bazy w tym przypadku?

Reply:
Jeśli dobrze rozumiem to chciałeś zebym zrobił dla klasy Entity funkcję Delete() która ustawia status IsDeleted na true zamiast robić tak jak jest teraz. No a ja po sprawdzeniu uważam z tego co robiłem research, że nie da się tak, bez wyciągania tego z bazy, dlatego chcialbym zostawić to tak jak jest.

Reply:
rzeczywiście nie da sie zrobić tego bez wyciągania z bazy
ale jak sie przyjrzysz to ty dużo wcześniej masz te dane wyciągane 
tutaj w argumencie nie dostajesz ID tylko właśnie ten obiekt

Reply:
Prawda, ale i tak UpdateOne/Async wymaga dwóch argumentów 
![mdb](https://user-images.githubusercontent.com/47067171/98711251-0ebfb780-2385-11eb-9cb1-a732390e6964.png)
Musiałbym ewentualnie usuwać poprzedni i dodawać nowy ale no wiadomo że tak się nie opłaca raczej.

Reply:
dobra niech już tak zostanie z tym filter + updateDefinition